infrastructure/lib/static-website.ts

```diff
@@ -0,0 +1,216 @@
+import {Construct, RemovalPolicy} from '@aws-cdk/core';
+import {Bucket} from '@aws-cdk/aws-s3';
+import {BucketDeployment, Source} from '@aws-cdk/aws-s3-deployment';
+import {DnsValidatedCertificate} from '@aws-cdk/aws-certificatemanager';
+import {AaaaRecord, ARecord, HostedZone, RecordTarget} from '@aws-cdk/aws-route53';
+import {BucketWebsiteTarget, CloudFrontTarget} from '@aws-cdk/aws-route53-targets';
+import {
+  CfnCloudFrontOriginAccessIdentity,
+  CfnDistribution,
+  ViewerProtocolPolicy,
+  PriceClass,
+  HttpVersion,
+  CloudFrontWebDistributionProps,
+  CloudFrontAllowedMethods,
+  SourceConfiguration,
+  CloudFrontWebDistribution,
+  Behavior
+} from '@aws-cdk/aws-cloudfront';
+import {CanonicalUserPrincipal, PolicyStatement, AccountRootPrincipal} from '@aws-cdk/aws-iam';
+
+/**
+ * Properties to configure the static website construct.
+ */
+export interface StaticWebsiteProps {
+  /**
+   * The hosted zone for the static website, e.g. example.com
+   * @default no DNS; required with domainName
+   */
+  hostedZone?: string;
+
+  /**
+   * The domain name for the static website, e.g. test.example.com
+   * @default no DNS; required with hostedZone
+   */
+  domainName?: string;
+
+  /**
+   * Where the artifacts for the website are stored.
+   */
+  artifactSourcePath: string;
```

Comment:
From experience people will eventually also want to be able to point to a deployed artifact (i.e. an S3 bucket/key). Use the `lambda.Code` pattern (something like `Source.fromAsset`, `Source.fromBucket`, etc). Even if you just implement one right now. Otherwise it will be a breaking change in the future.